flow/probes/ebpf.go

```diff
@@ -394,50 +111,31 @@ func (p *EBPFProbe) run() {
 				}
 			}
 
-			kernFlow := C.struct_flow{}
-			var key, nextKey C.__u64
+			var flowsRead int
 			for {
-				found, err := p.module.LookupNextElement(p.fmap, unsafe.Pointer(&key), unsafe.Pointer(&nextKey), unsafe.Pointer(&kernFlow))
+				found, err := p.module.LookupNextElement(p.fmap, unsafe.Pointer(&key), unsafe.Pointer(&nextKey), unsafe.Pointer(&kernFlows[nextAvailablePtr]))
 				if !found || err != nil {
+					p.module.DeleteElement(p.fmap, unsafe.Pointer(&key))
+					key = 0
 					break
 				}
-				key = nextKey
 				// delete every entry after we read the entry value
 				p.module.DeleteElement(p.fmap, unsafe.Pointer(&key))
+				key = nextKey
```

Comment:
we need to delete the previous key and not the new one, since deleting the new one will result in looking for a non existant key next time which will never fail but will always take us back to the beginning of the table (according to the bpf man) and therefore the loop will never end in some scenarios and also we will be starving flows